### PR TITLE
M3-5400: EU contract functionality for Linode create

### PR DESCRIPTION
**Pull request by HanaXu**
## Description
Add functionality for the EU Contract on Linode Create.

- The component should only be visible if the user has not signed the EU MC contract and is not a restricted user (restricted users have no access to view agreements)
- The "Create Linode" button should be disabled unless the the EU MC confirmation is checked.
- After the EU MC confirmation has been checked, when the user clicks "Create Linode", a POST to /account/agreements signing the eu model contract should be made. 

TODO:
- [x] Mutate React Query Cache when agreement is signed (wait for [7900](https://github.com/linode/manager/pull/7900) to get merged)

## How to test
On an account that hasn't signed the eu agreement:
- Go to `/linodes/create`
- Select a non-EU region:
    - You should not see the EU agreement checkbox in the checkout bar
- Select an EU region:
   - You should see the EU agreement checkbox and the `Create Linode` button should be disabled until it is checked
- Check the checkbox and click `Create Linode`
   - You should see a POST request made to `/account/agreements`
   - If there was an error POSTing to the agreements endpoint, you should see an error toast
   - If there was an error POSTing to `/linode/instances` after a successful POST to agreements (e.g. on dev bc there is no availability in the EU region), the EU checkbox should no longer display

Restricted users should not be able to see the EU agreement checkbox regardless of whether the parent account has signed or not signed the agreement and should be able to create a Linode as normal.

### CODE REVIEW

**Review by bnussman — APPROVED** [review]
Looks good to me. I'm going to take some things you did such as
- await agreement signing before making the api request to make the entity
- block entity creation if signing the agreement fails
- give user a visual error if the agreement api call fails

and apply them to my PR.

One minor thing we both didn't cover was un-checking the agreement if the user switched to another region and then back to an EU region, but I don't think that is necessary.

Like you said, use the React Query Cache Mutation so we don't need to refetch the endpoint once that work is merged and available

File: packages/manager/src/features/linodes/LinodesCreate/LinodeCreateContainer.tsx [hunk 1]
```diff
@@ -439,7 +471,24 @@ class LinodeCreateContainer extends React.PureComponent<CombinedProps, State> {
...
+        queryClient.invalidateQueries(queryKey);
+        this.setState({
+          showAgreement: false,
+        });
+      } catch (err) {
+        return this.props.enqueueSnackbar('Error signing agreement.', {
+          variant: 'error',
+        });
+      }
+    }
```
Comment (bnussman) [inline packages/manager/src/features/linodes/LinodesCreate/LinodeCreateContainer.tsx:490]: Oops! Forgot something. Should we do
```ts
this.setState({ formIsSubmitting: true });
```
so we have some loading state for this API call. Remember to
```ts
this.setState({ formIsSubmitting: false });
```
if the API gives an error for POSTing the agreement data.

Comment (bnussman) [reply packages/manager/src/features/linodes/LinodesCreate/LinodeCreateContainer.tsx:490]: Helper code for testing this 

```ts
  rest.post('*/account/agreements', async (req, res, ctx) => {
    const delay = (ms: number) => new Promise(res => setTimeout(res, ms));
    await delay(2000);
    return res(ctx.status(500), ctx.json({ reason: 'Unknown error' }));
  }),
  rest.get('*/account/agreements', (req, res, ctx) =>
    res(ctx.json({ eu_modal: false, privacy_policy: true }))
  ),
```

File: packages/manager/src/features/linodes/LinodesCreate/LinodeCreateContainer.tsx [hunk 2]
```diff
@@ -439,7 +471,24 @@ class LinodeCreateContainer extends React.PureComponent<CombinedProps, State> {
     return getLabel(arg1, arg2, arg3);
   };
 
-  submitForm: HandleSubmit = (_payload, linodeID?: number) => {
+  submitForm: HandleSubmit = async (_payload, linodeID?: number) => {
+    if (this.state.showAgreement) {
```
Comment (bnussman) [inline packages/manager/src/features/linodes/LinodesCreate/LinodeCreateContainer.tsx:475]: Logicly, does it make sense to make the API call if the Checkbox is showing, or if they checked the box? I chose to say `if(signedAgreement)` but I think we are accomplishing the same thing

Comment (bnussman) [reply packages/manager/src/features/linodes/LinodesCreate/LinodeCreateContainer.tsx:475]: Made this change and changed default `signedAgreement` state value to false (was the boolean value from the API)

File: packages/manager/src/features/linodes/LinodesCreate/LinodeCreateContainer.tsx (hunk 1 above)
Comment (bnussman) [reply packages/manager/src/features/linodes/LinodesCreate/LinodeCreateContainer.tsx:490]: Added proper loading state

File: packages/manager/src/features/linodes/LinodesCreate/LinodeCreateContainer.tsx [hunk 3]
```diff
@@ -457,14 +489,19 @@ class LinodeCreateContainer extends React.PureComponent<CombinedProps, State> {
...
+            errors: [
+              {
+                field: 'root_pass',
+                reason: passwordError,
+              },
+            ],
+          },
+          () => {
+            scrollErrorIntoView();
+          }
```
Comment (bnussman) [inline packages/manager/src/features/linodes/LinodesCreate/LinodeCreateContainer.tsx:503]: Added this as a minor UX fix, make sure this is ok!

**Review by DevDW — COMMENTED** [review]
Looks good, the only issue I observed was that with the mocks on and code in the MSW so that POSTs to `/account/agreements` fail, I never saw an error toast

**Comment by bnussman** [comment]
@DevDW Now that we are POSTing after successful entity creation I think we will just silently fail if the agreement request fails.

**Review by DevDW — APPROVED** [review]

**Pull request merged**